### PR TITLE
Update Creating-releases.md [skip ci]

### PR DESCRIPTION
**Pull request by michaelbadcrumble**
Updated this page to have some relevant info plus added more documentation.

### CODE REVIEW

File: docs/markdown/Creating-releases.md [hunk 1]
```diff
@@ -5,27 +5,47 @@ short-description: Creating releases
 # Creating releases
 
 In addition to development, almost all projects provide periodical
-source releases. These are standalone packages (usually either in tar
-or zip format) of the source code. They do not contain any revision
-control metadata, only the source code.
+source releases. These are standalone packages (usually either in
+tar or zip format) of the source code. They do not contain any
+revision control metadata, only the source code.  Meson provides
+a simple way of generating these, with the meson dist command.
```
Comment (TheQwertiest) [inline docs/markdown/Creating-releases.md:11]:
```suggestion
a simple way of generating these, with the `meson dist` command.
```

File: docs/markdown/Creating-releases.md [hunk 2]
```diff
@@ -5,27 +5,47 @@ short-description: Creating releases
...
-(especially the one tagged for the release).
+tree subdirectory `meson-dist`. This archive contains the full contents
+of the latest commit in revision control including all the submodules
+(recursively). All revision control metadata is removed. Meson then
+takes this archive and tests that it works by doing a full 
+`compile` + `test` + `install` cycle. If all these pass, Meson will
+then create a `SHA-256` checksum file next to the archive.
+
+
+## Autotools dist VS Meson build dist
```
Comment (TheQwertiest) [inline docs/markdown/Creating-releases.md:26]:
```suggestion
## Autotools dist VS Meson dist
```

**Comment by michaelbadcrumble** [comment]
Ping

File: docs/markdown/Creating-releases.md (hunk 1 above)
Comment (michaelbadcrumble) [reply docs/markdown/Creating-releases.md]: Highlights had been added to the appropriate command.

File: docs/markdown/Creating-releases.md (hunk 2 above)
Comment (michaelbadcrumble) [reply docs/markdown/Creating-releases.md]: Removed unrequited ward from deduction.

**Comment by TheQwertiest** [comment]
FYI: It often takes some time before PRs are accepted and merged (e.g. several weeks).
And just in case there was any misunderstanding - I'm just a fellow meson user and not a meson dev =)

**Comment by michaelbadcrumble** [comment]
That’s fine I’m going to be working on other projects anyway. Thanks.

**Comment by jpakkane** [comment]
[Don't do merges from master to your release branch](https://mesonbuild.com/Contributing.html#keeping-pull-requests-up-to-date). Use rebase instead. And it is not sufficient to have `[skip ci]` in the merge request title in Github, it has to be in the commit message itself.

**Comment by jpakkane** [comment]
I can squash this one, just be aware of this in the future.

**Pull request merged**